modules/how-to-plan-your-environment-according-to-application-requirements.adoc

```diff
@@ -69,3 +69,33 @@ do not. Most Java applications and applications that use huge pages are examples
 of applications that would not allow for overcommitment. That memory can not be
 used for other applications. In the example above, the environment would be
 roughly 30 percent overcommitted, a common ratio.
+
+The application pods can access a service either using environment variables which 
+gets injected by the Kubelet when a pod is run on a node for each active service or DNS. 
+A cluster-aware DNS server watches the Kubernetes API for new services and creates a set 
+of DNS records for each one. If DNS has been enabled throughout your cluster then all pods 
+should automatically be able to resolve services by their DNS name. Service discovery using 
+DNS can be used in case there is a need to go beyond 5000 services as the pods and deployments 
+will start failing due to argument list being too long after 5000 services in a namespace when 
+using environments variables for service discovery. It can be done by disabling the service 
```

Comment:
Disable the service links in the deployment's service specification file:


    ----
     - kind: Service
          apiVersion: v1
          metadata:
            name: service${IDENTIFIER}
          spec:
            enableServiceLinks: false
            selector:
              name: replicationcontroller${IDENTIFIER}
            ports:
            - name: serviceport${IDENTIFIER}
              protocol: TCP
              port: 80
              targetPort: 8080
            portalIP: ''
            type: ClusterIP
            sessionAffinity: None
          status:
            loadBalancer: {}
    ----